### DISCUSSION TITLE
github: pre-release action

### DISCUSSION THREAD

**Discussion by mkmkme** [finit-project/finit#350] [Ideas]
Hi Joachim!

I noticed you've been using a github action to automatically create a new release when the tag is pushed. I liked this implementation so much that I integrated it into [my repo](https://github.com/mkmkme/OpenGTA/pull/5) as well.

I noticed that in order to create an urgent release you made a change into that action. I also assume that you marked the `-rc1` as prerelease and wrote the description manually. In my implementation, I made two different actions: one looks for tags that are releases (`[0-9]+.[0-9]+.[0-9]+`) and one for pre-releases (`[0-9]+.[0-9]+.[0-9]+-*`). I chose the latter regex to comply with [semver rule about pre-releases](https://semver.org/#spec-item-9). I couldn't come up with a good approach for a release description there, so I picked tag description by default.

I thought it might be good to share an improvement of a schema that I borrowed in the first place :) Would you be interested if I made a similar pull request for finit as well?

**Comment by troglobit** [comment]
Glad you found it useful! :smiley: 

Yeah it would be really great to have a pre-release action as well, thank you for offering!

**Comment by mkmkme** [comment]
Hey @troglobit !

I'm terribly sorry for falling out of the loop with this one 😅 
I'll publish the PR shortly!

**Reply by mkmkme** [reply]
Oh wait, you've done it yourself already. Nevermind, sorry for being late!
